src/main/java/ch/njol/skript/expressions/ExprSpectatorTarget.java

```diff
@@ -24,60 +24,151 @@
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
-import ch.njol.skript.classes.Changer;
-import ch.njol.skript.doc.NoDoc;
-import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import com.destroystokyo.paper.event.player.PlayerStartSpectatingEntityEvent;
+import com.destroystokyo.paper.event.player.PlayerStopSpectatingEntityEvent;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.data.DefaultChangers;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.effects.Delay;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
 
-@NoDoc
-public class ExprSpectatorTarget extends SimplePropertyExpression<Player, Entity> {
+@Name("Spectator Target")
+@Description("Grabs the spectator target entity of the players.")
+@Examples({
+	"on player start spectating of player:",
+		"\tmessage \"&c%spectator target% currently has %{game::kills::%spectator target%}% kills!\" to the player",
+	"",
+	"on player stop spectating:",
+		"\tpast spectator target was a zombie",
+		"\tset spectator target to the nearest skeleton"
+})
+@RequiredPlugins("Paper")
+@Since("2.4-alpha4, INSERT VERSION (Paper Spectator Event)")
+public class ExprSpectatorTarget extends SimpleExpression<Entity> {
+
+	private static final boolean EVENT_SUPPORT = Skript.classExists("com.destroystokyo.paper.event.player.PlayerStartSpectatingEntityEvent");
 
 	static {
-		register(ExprSpectatorTarget.class, Entity.class, "spectator target", "players");
+		Skript.registerExpression(ExprSpectatorTarget.class, Entity.class, ExpressionType.PROPERTY,
+				"spectator target [of %-players%]", "%players%'[s] spectator target");
 	}
 
+	private Expression<Player> players;
+
 	@Override
-	protected String getPropertyName() {
-		return "spectator target";
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		players = (Expression<Player>) expressions[0];
+		if (players == null && !EVENT_SUPPORT) {
+			Skript.error("Your server platform does not support using 'spectator target' without players defined." +
+					"'spectator target of event-player'");
+			return false;
```

Comment:
This is just for the events right (so you can use `spectator target`)? If so, I think it would be better to just use a ParserInstance check here (isCurrentEvent)

Reply:
The paper events may not exist. That'll throw an error checking a class against a class that doesn't exist.